third_party/move/move-model/src/ast.rs

```diff
@@ -959,17 +959,17 @@ impl ExpData {
     ///
     /// In every case, if `visitor` returns `false`, then the visit is stopped early; otherwise
     /// the the visit will continue.
-    pub fn visit_positions<F>(&self, visitor: &mut F)
+    pub fn visit_positions<F>(&self, visitor: &mut F) -> Option<()>
```

Comment:
On second thought, you don't really need `visit_positions` since your visitor only cares about `Pre` position.  There is already a function `ExpData::any` which visits pre-order and returns `true` if visitor ever returns `true`.  You can negate the results to find out if any returns `false`.

Reply:
Done.
It's better to return a bool for the interface, but `Option<T>` has a clear meaning, which is a computation that returns `T` but may fail (returns `Some(t)` on success and `None` on failure).
I don't want to use `any` because it's uncommon for a predicate function to be stateful. But it's common to have stateful visitor functions, as we often accumulate contexts (say scopes) when visiting sub-expressions.